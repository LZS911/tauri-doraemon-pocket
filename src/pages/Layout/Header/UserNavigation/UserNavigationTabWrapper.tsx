import { UserNavigationTabWrapperProps } from '.';

const UserNavigationTabWrapper: React.FC<UserNavigationTabWrapperProps> = ({
  dataSource,
}) => {
  return (
    <div className="flex flex-col">
      {dataSource.map((v) => {
        return (
          !v.hidden && (
            <div
              key={v.text}
              onClick={v.onClick}
              className="flex cursor-pointer items-center rounded-sm px-4 py-2 transition-[background] hover:bg-slate-200 dark:text-white dark:hover:bg-slate-500"
            >
              {v.icon}
              <div className="ml-4">{v.text}</div>
            </div>
          )
        );
      })}
    </div>
  );
};

export default UserNavigationTabWrapper;
