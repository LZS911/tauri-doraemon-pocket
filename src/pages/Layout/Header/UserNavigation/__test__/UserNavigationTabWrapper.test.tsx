import { fireEvent, render, screen } from '@testing-library/react';
import { ITabWrapperDataSource } from '..';
import UserNavigationTabWrapper from '../UserNavigationTabWrapper';

const onClick = jest.fn();

const dataSource: ITabWrapperDataSource[] = [
  {
    icon: <>icon1</>,
    onClick: onClick,
    text: 'text1',
  },
  {
    icon: <>icon2</>,
    onClick: jest.fn(),
    text: 'text2',
    hidden: true,
  },
];

describe('test UserNavigationTabWrapper', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <UserNavigationTabWrapper dataSource={dataSource} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should be executed onClick when click icon', () => {
    render(<UserNavigationTabWrapper dataSource={dataSource} />);
    expect(onClick).toBeCalledTimes(0);

    fireEvent.click(screen.getByText('icon1'));
    expect(onClick).toBeCalledTimes(1);
  });
});
