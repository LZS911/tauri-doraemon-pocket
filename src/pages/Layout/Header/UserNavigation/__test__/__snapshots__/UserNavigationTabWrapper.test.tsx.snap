// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserNavigationTabWrapper should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="flex cursor-pointer items-center rounded-sm px-4 py-2 transition-[background] hover:bg-slate-200 dark:text-white dark:hover:bg-slate-500"
    >
      icon1
      <div
        class="ml-4"
      >
        text1
      </div>
    </div>
  </div>
</div>
`;
